core/config/value.go

```diff
@@ -487,8 +486,26 @@ func (cv Value) getValueStruct(key string, target interface{}) (interface{}, boo
 			if destSlice.Len() > 0 {
 				fieldValue.Set(destSlice)
 			}
-		}
+		case bucketMap:
+			createMap := reflect.MakeMap(fieldType).Interface()
+			val := global.GetValue(childKey).Value()
 
+			if val != nil {
+				switch v := val.(type) {
+				case map[interface{}]interface{}:
+					destMap := createMap.(map[string]interface{})
```

Comment:
Is this type assertion always going to be correct? This is the case for `reflect.Map` but it could be a map of anything? What if the type is some other type of map? Can we add some tests around this?